src/components/Calculator.js

```diff
@@ -1,55 +1,45 @@
 /* eslint-disable */
 
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  handleClick = (buttonName) => {
-    this.setState((state) => calculate(state, buttonName));
-  };
-
-  showResult = () => {
-    const { total, next, operation } = this.state;
-    const show = (total || '') + (operation || '') + (next || '');
-
-    return (show || '0');
+const Calculator = () => {
+  const [state, setState] = useState({});
+  const handleClick = (ev) => {
```

Comment:
- [ ] **[Optional] The application crashes whenever I click any operator multiple times. Please consider fixing this as it will improve the user experience. Also, error handling should be one of the primary concerns.**

![operatordouble](https://user-images.githubusercontent.com/59168430/133877602-302e5595-4849-4a8a-93da-58d2a48c42c6.gif)

Reply:
Your note is correct - I will fix it soon. Thanks